### PR TITLE
NettyDockerCmdExecFactory ignores API version configuration

### PR DESCRIPTION
**Pull request by jamesnetherton**
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/808)

### CODE REVIEW

File: src/test/java/com/github/dockerjava/netty/NettyDockerCmdExecFactoryConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    @Test
+    public void testNettyDockerCmdExecFactoryConfigWithoutApiVersion() throws Exception {
+        NettyDockerCmdExecFactory factory = new NettyDockerCmdExecFactory();
+        Builder configBuilder = DefaultDockerClientConfig.createDefaultConfigBuilder().withDockerHost("tcp://localhost:2378");
+
+        DockerClient client = DockerClientBuilder.getInstance(configBuilder)
+            .withDockerCmdExecFactory(factory)
+            .build();
+
+        FakeDockerServer server = new FakeDockerServer(2378);
```
Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/netty/NettyDockerCmdExecFactoryConfigTest.java]: Please use some random port either it may get conflicts.

Comment (jamesnetherton) [inline src/test/java/com/github/dockerjava/netty/NettyDockerCmdExecFactoryConfigTest.java]: Thanks - I just pushed some mods to ensure a random port is used.

**Comment by jamesnetherton** [comment]
@KostyaSha any chance of getting this merged?

**Comment by jamesnetherton** [comment]
Thanks!

**Pull request merged**